Sources/LiveKit/types/VideoParameters.swift

```diff
@@ -1,6 +1,32 @@
 import Foundation
 import WebRTC
 
+extension Collection where Element == VideoParameters {
+
+    func suggestedPresetIndex(dimensions: Dimensions? = nil,
+                              videoEncoding: VideoEncoding? = nil) -> Int {
+        // self must at lease have 1 element
+        assert(!isEmpty)
+        // dimensions or videoEndocing is required
+        assert(dimensions != nil || videoEncoding != nil)
+
+        var result = 0
+        for preset in self {
+            if let dimensions = dimensions,
```

Comment:
There was a similar utility routine ☝️ to search presets. Maybe use that to avoid repetition and keep logic in one place?